mycore-base/src/main/java/org/mycore/common/content/MCRSourceContent.java

```diff
@@ -117,6 +70,52 @@ public MCRSourceContent(Source source) {
         this.setBaseContent(baseContent);
     }
 
+    private static MCRContent getBaseContent(StreamSource streamSource) {
+        InputStream inputStream = streamSource.getInputStream();
+        if (inputStream != null) {
+            return new MCRStreamContent(inputStream);
+        } else {
+            try {
+                URI uri = new URI(streamSource.getSystemId());
+                return new MCRURLContent(uri.toURL());
+            } catch (URISyntaxException | MalformedURLException e) {
+                throw new MCRException("Could not create instance of MCRURLContent for SYSTEMID: "
+                    + streamSource.getSystemId(), e);
+            }
+        }
+    }
+
+    private static MCRByteContent getBaseContent(JAXBSource source) {
+        TransformerFactory transformerFactory = TransformerFactory.newDefaultInstance();
+        try {
+            Transformer transformer = transformerFactory.newTransformer();
+            ByteArrayOutputStream bout = new ByteArrayOutputStream();
+            transformer.transform(source, new StreamResult(bout));
+            return new MCRByteContent(bout.toByteArray());
+        } catch (TransformerException e) {
+            throw new MCRException("Error while resolving JAXBSource", e);
+        }
+    }
+
+    private static MCRJDOMContent getBaseContent(JDOMSource src) {
+        Document xml = src.getDocument();
+        if (xml != null) {
+            return new MCRJDOMContent(xml);
+        }
+        return src.getNodes().stream()
+            .filter(Parent.class::isInstance) //only Element or Document matter
+            .map(Parent.class::cast)
+            .findFirst()
+            .map(p -> switch (p){
```

Comment:
Looks correct. This method is still very confusing though. In one case you want the document, in another the element and in the third a cloned element.

Reply:
It's not what I want, but what I get. The resulting content should not return more "content" than that is returned by the Source. These are all cases that occurred in the wild and had to be handled like this. The amount of cases is mainly to reduce the work that has to be done to get the proper result. One could have done a simple "clone()" on Element to have a detached copy of the given element, but this involves a copy every time even if it's not really needed.